### PR TITLE
usb_hotplug_unplug: add hotplug/hotunplug support for usb-storage

### PR DESCRIPTION
**Pull request by ybduan**
1.Add clean dmesg step before test in format_disk
2.Add clean partition step at the end of test in format_disk
3.Add hotplug/hotunplug support for usb-storage

ID: 1802011
Signed-off-by: ybduan <yduan@redhat.com>

### CODE REVIEW

**Comment by ybduan** [comment]
Depends on https://github.com/avocado-framework/avocado-vt/pull/2490

**Comment by ybduan** [comment]
'01-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.usb.usb_storage.without_usb_hub.piix3-uhci' Finshed 2020-03-25 06:03:48 2020-03-25 06:09:25 PASS 336.901971
'02-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.usb.usb_storage.without_usb_hub.piix4-uhci' Finshed 2020-03-25 06:09:25 2020-03-25 06:14:57 PASS 331.244097
'03-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.usb.usb_storage.without_usb_hub.ich9-uhci' Finshed 2020-03-25 06:14:57 2020-03-25 06:20:25 PASS 328.8418
'04-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.usb.usb_storage.without_usb_hub.ich9-ehci' Finshed 2020-03-25 06:20:25 2020-03-25 06:24:01 PASS 215.127328
'05-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.usb.usb_storage.without_usb_hub.usb-ehci' Finshed 2020-03-25 06:24:01 2020-03-25 06:27:32 PASS 211.287058
'06-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.usb.usb_storage.without_usb_hub.nec-xhci' Finshed 2020-03-25 06:27:32 2020-03-25 06:31:06 PASS 214.268104
'07-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.usb.usb_storage.without_usb_hub.qemu-xhci' Finshed 2020-03-25 06:31:06 2020-03-25 06:34:39 PASS 212.633478
'08-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.usb.usb_storage.hotplug_unplug_1.without_usb_hub.piix3-uhci' Finshed 2020-03-25 06:34:39 2020-03-25 06:37:12 PASS 152.825371
'09-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.usb.usb_storage.hotplug_unplug_1.without_usb_hub.piix4-uhci' Finshed 2020-03-25 06:37:12 2020-03-25 06:39:43 PASS 151.099378
'10-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.usb.usb_storage.hotplug_unplug_1.without_usb_hub.ich9-uhci' Finshed 2020-03-25 06:39:43 2020-03-25 06:42:12 PASS 149.252623
'11-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.usb.usb_storage.hotplug_unplug_1.without_usb_hub.ich9-ehci' Finshed 2020-03-25 06:42:12 2020-03-25 06:42:50 PASS 38.039782
'12-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.usb.usb_storage.hotplug_unplug_1.without_usb_hub.usb-ehci' Finshed 2020-03-25 06:42:50 2020-03-25 06:43:27 PASS 36.698292
'13-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.usb.usb_storage.hotplug_unplug_1.without_usb_hub.nec-xhci' Finshed 2020-03-25 06:43:27 2020-03-25 06:44:00 PASS 32.825537
'14-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.usb.usb_storage.hotplug_unplug_1.without_usb_hub.qemu-xhci' Finshed 2020-03-25 06:44:00 2020-03-25 06:44:32 PASS 32.666035
'15-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.usb.usb_storage.hotplug_unplug_300.without_usb_hub.ich9-ehci' Finshed 2020-03-25 06:44:32 2020-03-25 06:46:29 PASS 116.889826
'16-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.usb.usb_storage.hotplug_unplug_300.without_usb_hub.usb-ehci' Finshed 2020-03-25 06:46:29 2020-03-25 06:48:24 PASS 114.520632
'17-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.usb.usb_storage.hotplug_unplug_300.without_usb_hub.nec-xhci' Finshed 2020-03-25 06:48:24 2020-03-25 06:50:04 PASS 100.047363
'18-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.usb.usb_storage.hotplug_unplug_300.without_usb_hub.qemu-xhci' Finshed 2020-03-25 06:50:04 2020-03-25 06:51:46 PASS 102.26689
Summary: 
Finshed=18, PASS=18

**Comment by ybduan** [comment]
'01-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.usb.usb_storage.without_usb_hub.piix3-uhci' Finshed 2020-03-25 06:57:33 2020-03-25 06:59:23 PASS 109.537617
'02-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.usb.usb_storage.without_usb_hub.ich9-uhci' Finshed 2020-03-25 06:59:23 2020-03-25 07:00:55 PASS 92.667461
'03-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.usb.usb_storage.without_usb_hub.ich9-ehci' Finshed 2020-03-25 07:00:55 2020-03-25 07:07:27 PASS 391.552274
'04-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.usb.usb_storage.without_usb_hub.usb-ehci' Finshed 2020-03-25 07:07:27 2020-03-25 07:08:21 PASS 54.234271
'05-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.usb.usb_storage.without_usb_hub.nec-xhci' Finshed 2020-03-25 07:08:21 2020-03-25 07:09:06 PASS 44.581148
'06-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.usb.usb_storage.without_usb_hub.qemu-xhci' Finshed 2020-03-25 07:09:06 2020-03-25 07:09:50 PASS 44.294566
'07-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.usb.usb_storage.hotplug_unplug_1.without_usb_hub.piix3-uhci' Finshed 2020-03-25 07:09:50 2020-03-25 07:11:20 PASS 89.447318
'08-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.usb.usb_storage.hotplug_unplug_1.without_usb_hub.ich9-uhci' Finshed 2020-03-25 07:11:20 2020-03-25 07:12:49 PASS 89.624304
'09-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.usb.usb_storage.hotplug_unplug_1.without_usb_hub.ich9-ehci' Finshed 2020-03-25 07:12:49 2020-03-25 07:13:44 PASS 55.043525
'10-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.usb.usb_storage.hotplug_unplug_1.without_usb_hub.usb-ehci' Finshed 2020-03-25 07:13:44 2020-03-25 07:20:17 PASS 392.584175
'11-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.usb.usb_storage.hotplug_unplug_1.without_usb_hub.nec-xhci' Finshed 2020-03-25 07:20:17 2020-03-25 07:21:02 PASS 45.375178
'12-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.usb.usb_storage.hotplug_unplug_1.without_usb_hub.qemu-xhci' Finshed 2020-03-25 07:21:02 2020-03-25 07:21:48 PASS 45.404155
'13-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.usb.usb_storage.hotplug_unplug_300.without_usb_hub.ich9-ehci' Finshed 2020-03-25 07:21:48 2020-03-25 07:25:36 PASS 228.009224
'14-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.usb.usb_storage.hotplug_unplug_300.without_usb_hub.usb-ehci' Finshed 2020-03-25 07:25:36 2020-03-25 07:35:00 PASS 563.832896
'15-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.usb.usb_storage.hotplug_unplug_300.without_usb_hub.nec-xhci' Finshed 2020-03-25 07:35:00 2020-03-25 07:37:02 PASS 122.003141
'16-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.usb.usb_storage.hotplug_unplug_300.without_usb_hub.qemu-xhci' Finshed 2020-03-25 07:37:02 2020-03-25 07:39:07 PASS 125.836368
Summary: 
Finshed=16, PASS=16

**Comment by ybduan** [comment]
Please help review this PR, thanks! @XueqiangWei

File: qemu/tests/cfg/usb.cfg [hunk 1]
```diff
@@ -266,6 +266,24 @@
                 check_serial_option = no
                 check_removable_option = no
                 check_io_size_option = no
+            variants:
+                - @default:
+                    with_hotplug_unplug = no
+                - hotplug_unplug_1:
```
Comment (XueqiangWei) [inline qemu/tests/cfg/usb.cfg]: Can we change the hotplug_unplug_1 to usb_one_time like line 295 to line 300 under usb_host
e.g.
variants:
         - usb_one_time:
               usb_repeat_times = 1
         - usb_multi_times:
               usb_repeat_times = 300

File: qemu/tests/cfg/usb.cfg [hunk 2]
```diff
@@ -266,6 +266,24 @@
...
+                    usb_repeat_times = 1
+                    with_hotplug_unplug = yes
+                    boot_drive_stg = no
+                    check_serial_option = no
+                    check_removable_option = no
+                    check_io_size_option = no
+                - hotplug_unplug_300:
```
Comment (XueqiangWei) [inline qemu/tests/cfg/usb.cfg]: same as above

File: qemu/tests/usb_storage.py [hunk 3]
```diff
@@ -193,56 +193,83 @@ def _check_io_size_option(min_io_size="512", opt_io_size="0"):
...
-        _check_removable_option(removable, expect_str)
-
-    if params.get("check_io_size_option") == "yes":
-        error_context.context("Check usb min/opt io_size option", logging.info)
-        _check_io_size_option("0", "0")
-        # Guest can't recognize correct value which we set now,
-        # So comment these test temporary.
-        # _check_io_size_option("1024", "1024")
-        # _check_io_size_option("4096", "4096")
+    hotplug_unplug = (params["with_hotplug_unplug"] == "yes")
```
Comment (XueqiangWei) [inline qemu/tests/usb_storage.py]: Please add doc string to line 14 for hotplug/unplug steps.

File: qemu/tests/cfg/usb.cfg (hunk 1 above)
Comment (ybduan) [reply qemu/tests/cfg/usb.cfg]: Updated, thanks!

File: qemu/tests/cfg/usb.cfg (hunk 2 above)
Comment (ybduan) [reply qemu/tests/cfg/usb.cfg]: Updated, thanks!

File: qemu/tests/usb_storage.py (hunk 3 above)
Comment (ybduan) [reply qemu/tests/usb_storage.py]: Updated, thanks!

**Comment by XueqiangWei** [comment]
Test all usb.usb_storage cases:
1. with rhel820  + virtio-scsi
Summary: 
Finshed=33, PASS=33
2. with rhel820, win10 + virtio-blk
Summary: 
Finshed=62, PASS=62

LGTM,
Acked-by: xuwei@redhat.com

**Comment by ybduan** [comment]
Please help review this PR, thanks! @lolyu

File: qemu/tests/usb_storage.py [hunk 4]
```diff
@@ -193,56 +195,83 @@ def _check_io_size_option(min_io_size="512", opt_io_size="0"):
...
+        # No bus specified, default using "usb.0" for "usb-storage"
+        for i in params["chk_usb_info_keyword"].split(","):
+            _verify_string(i, output, [i])
+        session.close()
+        _do_io_test_guest()
+
+        # this part is linux only
+        if params.get("check_serial_option") == "yes":
+            error_context.context("Check usb serial option", logging.info)
+            serial = str(uuid.uuid4())
```
Comment (lolyu) [inline qemu/tests/usb_storage.py]: better to use `uuid.uuid4().hex`

File: qemu/tests/usb_storage.py [hunk 5]
```diff
@@ -193,56 +195,83 @@ def _check_io_size_option(min_io_size="512", opt_io_size="0"):
...
+            logging.info("Check this option with some illegal string")
+            logging.info("Set usb serial to a empty string")
+            # An empty string, ""
+            serial = "EMPTY_STRING"
+            regex_str = r'usb-storage.*?serial = (.*?)\s'
+            _check_serial_option(serial, regex_str, '""')
+
+            logging.info("Leave usb serial option blank")
+            serial = "NO_EQUAL_STRING"
+            regex_str = r'usb-storage.*?serial = (.*?)\s'
```
Comment (lolyu) [inline qemu/tests/usb_storage.py]: is this `regex_str` same as LINE 245?

File: qemu/tests/usb_storage.py [hunk 6]
```diff
@@ -193,56 +195,83 @@ def _check_io_size_option(min_io_size="512", opt_io_size="0"):
...
+            _check_removable_option(removable, expect_str)
+
+            removable = "off"
+            expect_str = "Attached SCSI disk"
+            _check_removable_option(removable, expect_str)
+
+        if params.get("check_io_size_option") == "yes":
+            error_context.context("Check usb min/opt io_size option", logging.info)
+            _check_io_size_option("0", "0")
+            # Guest can't recognize correct value which we set now,
```
Comment (lolyu) [inline qemu/tests/usb_storage.py]: better to use a codetag `NOTE` here in comment.
Please refer to: https://www.python.org/dev/peps/pep-0350/#general-syntax

File: qemu/tests/usb_storage.py (hunk 4 above)
Comment (ybduan) [reply qemu/tests/usb_storage.py]: Updated, thanks!

File: qemu/tests/usb_storage.py (hunk 5 above)
Comment (ybduan) [reply qemu/tests/usb_storage.py]: Have deleted the duplicate one, Line250. Thanks!

File: qemu/tests/usb_storage.py (hunk 6 above)
Comment (ybduan) [reply qemu/tests/usb_storage.py]: Updated, thanks!

**Comment by lolyu** [comment]
please test it again.

**Comment by ybduan** [comment]
Test results of RHEL and Windows guests are all passed:

'01-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.usb.usb_storage.without_usb_hub.piix3-uhci' Finshed 2020-04-02 23:19:00 2020-04-02 23:24:30 PASS 330.51411
'02-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.usb.usb_storage.without_usb_hub.piix4-uhci' Finshed 2020-04-02 23:24:30 2020-04-02 23:30:01 PASS 330.366891
'03-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.usb.usb_storage.without_usb_hub.ich9-uhci' Finshed 2020-04-02 23:30:01 2020-04-02 23:35:28 PASS 327.55724
'04-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.usb.usb_storage.without_usb_hub.ich9-ehci' Finshed 2020-04-02 23:35:28 2020-04-02 23:39:03 PASS 214.655763
'05-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.usb.usb_storage.without_usb_hub.usb-ehci' Finshed 2020-04-02 23:39:03 2020-04-02 23:42:33 PASS 210.574598
'06-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.usb.usb_storage.without_usb_hub.nec-xhci' Finshed 2020-04-02 23:42:33 2020-04-02 23:46:01 PASS 207.584762
'07-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.usb.usb_storage.without_usb_hub.qemu-xhci' Finshed 2020-04-02 23:46:01 2020-04-02 23:49:33 PASS 211.461432
'08-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.usb.usb_storage.hotplug_unplug_one_time.without_usb_hub.piix3-uhci' Finshed 2020-04-02 23:49:33 2020-04-02 23:52:08 PASS 155.570622
'09-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.usb.usb_storage.hotplug_unplug_one_time.without_usb_hub.piix4-uhci' Finshed 2020-04-02 23:52:08 2020-04-02 23:54:38 PASS 149.386721
'10-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.usb.usb_storage.hotplug_unplug_one_time.without_usb_hub.ich9-uhci' Finshed 2020-04-02 23:54:38 2020-04-02 23:57:06 PASS 148.213886
'11-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.usb.usb_storage.hotplug_unplug_one_time.without_usb_hub.ich9-ehci' Finshed 2020-04-02 23:57:06 2020-04-02 23:57:42 PASS 36.41501
'12-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.usb.usb_storage.hotplug_unplug_one_time.without_usb_hub.usb-ehci' Finshed 2020-04-02 23:57:42 2020-04-02 23:58:20 PASS 37.616591
'13-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.usb.usb_storage.hotplug_unplug_one_time.without_usb_hub.nec-xhci' Finshed 2020-04-02 23:58:20 2020-04-02 23:58:56 PASS 36.127314
'14-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.usb.usb_storage.hotplug_unplug_one_time.without_usb_hub.qemu-xhci' Finshed 2020-04-02 23:58:56 2020-04-02 23:59:29 PASS 32.756679
'15-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.usb.usb_storage.hotplug_unplug_multi_times.without_usb_hub.ich9-ehci' Finshed 2020-04-02 23:59:29 2020-04-03 00:43:01 PASS 2612.022788
'16-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.usb.usb_storage.hotplug_unplug_multi_times.without_usb_hub.usb-ehci' Finshed 2020-04-03 00:43:01 2020-04-03 01:26:00 PASS 2579.562698
'17-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.usb.usb_storage.hotplug_unplug_multi_times.without_usb_hub.nec-xhci' Finshed 2020-04-03 01:26:00 2020-04-03 02:00:56 PASS 2095.807692
'18-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.usb.usb_storage.hotplug_unplug_multi_times.without_usb_hub.qemu-xhci' Finshed 2020-04-03 02:00:56 2020-04-03 02:35:52 PASS 2095.582291
'19-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.usb.usb_storage.without_usb_hub.piix3-uhci' Finshed 2020-04-03 02:35:52 2020-04-03 02:37:23 PASS 91.338934
'20-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.usb.usb_storage.without_usb_hub.ich9-uhci' Finshed 2020-04-03 02:37:23 2020-04-03 02:38:53 PASS 89.558113
'21-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.usb.usb_storage.without_usb_hub.ich9-ehci' Finshed 2020-04-03 02:38:53 2020-04-03 02:45:24 PASS 391.139853
'22-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.usb.usb_storage.without_usb_hub.usb-ehci' Finshed 2020-04-03 02:45:24 2020-04-03 02:51:55 PASS 391.537829
'23-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.usb.usb_storage.without_usb_hub.nec-xhci' Finshed 2020-04-03 02:51:55 2020-04-03 02:52:40 PASS 44.206557
'24-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.usb.usb_storage.without_usb_hub.qemu-xhci' Finshed 2020-04-03 02:52:40 2020-04-03 02:53:24 PASS 44.339457
'25-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.usb.usb_storage.hotplug_unplug_one_time.without_usb_hub.piix3-uhci' Finshed 2020-04-03 02:53:24 2020-04-03 02:54:54 PASS 89.856482
'26-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.usb.usb_storage.hotplug_unplug_one_time.without_usb_hub.ich9-uhci' Finshed 2020-04-03 02:54:54 2020-04-03 02:56:24 PASS 89.602595
'27-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.usb.usb_storage.hotplug_unplug_one_time.without_usb_hub.ich9-ehci' Finshed 2020-04-03 02:56:24 2020-04-03 03:02:56 PASS 392.653352
'28-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.usb.usb_storage.hotplug_unplug_one_time.without_usb_hub.usb-ehci' Finshed 2020-04-03 03:02:56 2020-04-03 03:03:51 PASS 54.893095
'29-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.usb.usb_storage.hotplug_unplug_one_time.without_usb_hub.nec-xhci' Finshed 2020-04-03 03:03:51 2020-04-03 03:04:37 PASS 45.419566
'30-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.usb.usb_storage.hotplug_unplug_one_time.without_usb_hub.qemu-xhci' Finshed 2020-04-03 03:04:37 2020-04-03 03:05:22 PASS 45.463212
'31-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.usb.usb_storage.hotplug_unplug_multi_times.without_usb_hub.ich9-ehci' Finshed 2020-04-03 03:05:22 2020-04-03 04:43:16 PASS 5874.40061
'32-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.usb.usb_storage.hotplug_unplug_multi_times.without_usb_hub.usb-ehci' Finshed 2020-04-03 04:43:17 2020-04-03 06:25:48 PASS 6151.133618
'33-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.usb.usb_storage.hotplug_unplug_multi_times.without_usb_hub.nec-xhci' Finshed 2020-04-03 06:25:48 2020-04-03 07:11:58 PASS 2770.455311
'34-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.usb.usb_storage.hotplug_unplug_multi_times.without_usb_hub.qemu-xhci' Finshed 2020-04-03 07:11:58 2020-04-03 07:58:12 PASS 2773.817322
Summary: 
Finshed=34, PASS=34

**Pull request merged**